BiconeEvolution/current_antenna_evo_build/XF_Loop/Evolutionary_Loop/Antenna_Performance_Metric/newCombineErrors.py

```diff
@@ -0,0 +1,139 @@
+__author__ = 'Dylan Wells <wells.1629@osu.edu'
+
+# Imports
+import argparse
+import math
+from pathlib import Path
+
+import pandas as pd
+
+# Arguments
+desc = ("This script will combine the errors and fitness scores "
+        "of individuals made with reproduction")
+parser = argparse.ArgumentParser(description=desc)
+parser.add_argument("source", help="source directory for fitness files", type=Path)
+parser.add_argument("gen", help="Generation number.", type=int)
+parser.add_argument("NPOP", help="Number of individuals in the population.", type=int)
+
+g = parser.parse_args()
+
+source_dir = g.source
+curr_gen = g.gen
+prev_gen = g.gen - 1
+prev_DNA_file = source_dir / f'{prev_gen}_generationDNA.csv'
+curr_DNA_file = source_dir / f'{curr_gen}_generationDNA.csv'
+prev_fitness = source_dir / f'{prev_gen}_fitnessScores.csv'
+curr_fitness = source_dir / f'{curr_gen}_fitnessScores.csv'
+prev_errors = source_dir / f'{prev_gen}_errorBars.csv'
+curr_errors = source_dir / f'{curr_gen}_errorBars.csv'
+
+identical_indivs = []
+
+# Load in the DNA files from the previous and current generation
+# variables are S, H, Xi, Yi, Zf, Yf, beta 
+skiprows = 8
+
+def read_DNA(DNA_file):
+    '''Return a list of tuples containing the DNA of each individual in the population'''
+    DNA_csv = pd.read_csv(DNA_file, header=None, 
+                          delimiter=',', skiprows=skiprows)
+    DNA = {
+        "S": DNA_csv.iloc[:, 0].values,
+        "H": DNA_csv.iloc[:, 1].values,
+        "Xi": DNA_csv.iloc[:, 2].values,
+        "Yi": DNA_csv.iloc[:, 3].values,
+        "Zf": DNA_csv.iloc[:, 4].values,
+        "Yf": DNA_csv.iloc[:, 5].values,
```

Comment:
> Here is a sample output from testing on Generation 9 and 10 of the pueo loop:

Is that a list of individuals that were identical between generations? How many were there in total (looks like it might be cut off). Anyway, I think it sounds reasonable that we would get a decent number repeated, especially if the constraints are making things like crossover difficult (although then I guess they would be mutated?). I know that we had mutation set to 3 for that run, so we should expect at least 3 reproduced per generation.

Also, on this line, I think we have Zf and Yf backwards. We used to have it in this order, but currently the GA has Yf in the 5th column and Yf in the 4th (which I think is good practice). See here: https://github.com/osu-particle-astrophysics/Shared-Code/blob/36f5f58431004445ef6731b512c990e903f253ea/GA/HeaderFiles/GeneratePUEO.h#L49

Reply:
Yeah, so it shows that 21 of the 50 individuals we saw in generation 10 had at least one identical individual in generation 9, many had multiple. I don't know if this is similar to what we would see in the Bicone loop, but it seems very high to me.
And, ok, I made a commit to change the order of YF and ZF; it doesn't affect the output, but it's good to have consistency.